fs/span-manager/span_manager.go

```diff
@@ -161,14 +185,14 @@ func (m *SpanManager) GetContents(startUncompOffset, endUncompOffset compression
 				return err
 			}
 			spanReaders[j] = r
+			spanClosers[j] = r
 			return nil
 		})
 	}
 	if err := eg.Wait(); err != nil {
 		return nil, err
 	}
-
-	return io.MultiReader(spanReaders...), nil
+	return &MultiReaderCloser{spanClosers, io.MultiReader(spanReaders...)}, nil
```

Comment:
nit: to mimic the API of `io.MultiReader(spanReaders...)` would it make sense to add a `multiReadCloser(spanReadClosers...)` function here which basically just returns what you've written here?

This is definitely not blocking, just food for thought.

Reply:
We can but since I can't pass in a `[]io.ReadCloser` to `io.MultiReader` I'll have to either pass both slices or only pass the reader slice and have to loop again through the readers  in `multiReadCloser` to get a slice of `io.Closer` to pass to `MultiReaderCloser`.

Reply:
Yeah, this has the same problem as my previous comment. Feel free to ignore.